etc/init.d/dscpclassify

```diff
@@ -3,11 +3,92 @@
 START=20
 USE_PROCD=1
 
-lan="$(fw4 -q zone lan)"
-wan="$(fw4 -q zone wan)"
+log() {
+	logger -t dscpclassify -p "daemon.$1" "$2"
+}
+
+list_append() {
+	list="$list"$'\n'"$1"
+}
+
+config_foreach_reverse() {
+	local list
+
+	config_foreach list_append "$2"
+	list=$(echo "$list" | sort -r)
+
+	for config in $list; do
+		"$1" "$config" "$3"
+	done
+}
+
+config_get_uint() {
+	local err uint
+
+	err=0
+	config_get uint "$2" "$3" "$4"
+	if [ -n "$uint" ] && { ! [ "$uint" -eq "$uint" ] 2>/dev/null || [ "$uint" -lt 0 ]; }; then
+		err=1
+		uint=""
+	fi
+	eval "$1"="$uint"
+	return "$err"
+}
 
 mklist() {
-	echo "$*" | tr '\n' ' ' | sed -e 's/^\s*//' -e 's/\s*$//' -e 's/\s\+/, /g'
+	echo "$*" | tr '\n' ' ' | sed -e 's/^\s*//' -e 's/\s*$//' -e 's/\([^.]\)\s\+\([^.]\)/\1, \2/g'
+}
+
+check_class() {
+	local class
+
+	class="$(echo "$1" | tr 'A-Z' 'a-z')"
+	[ "$class" = "be" ] && class="cs0"
+
+	case "$class" in
+	cs0 | le | cs1 | af11 | af12 | af13 | cs2 | af21 | af22 | af23 | cs3 | af31 | af32 | af33 | cs4 | af41 | af42 | af43 | cs5 | va | ef | cs6 | cs7)
+		echo "$class"
+		return 0
+		;;
+	esac
+	return 1
+}
+
+check_duration() {
+	echo "$1" | grep -q -E -e "^[[:digit:]]+[smhd]$" -e "^[[:digit:]]+[smhd][[:digit:]]+[smhd]$" -e "^[[:digit:]]+[smhd][[:digit:]]+[smhd][[:digit:]]+[smhd]$" -e "^[[:digit:]]+[smhd][[:digit:]]+[smhd][[:digit:]]+[smhd][[:digit:]]+[smhd]$"
+}
+
+check_family() {
+	case $1 in
+	ipv4 | ipv6) return 0 ;;
+	esac
+	return 1
+}
+
+check_port() {
+	for i in $1; do
+		echo "$i" | grep -q -E -e "^[[:digit:]]+-[[:digit:]]+$" -e "^[[:digit:]]+$" || return 1
+	done
+}
+
+check_port_proto() {
+	for i in $1; do
+		case $i in
+		tcp | udp) true ;;
+		*) return 1 ;;
+		esac
+	done
+}
+
+sort_ip() {
+	for i in $1; do
+		case $i in
+		[0-9]*.[0-9]*.[0-9]*.*[0-9]) ipv4="$ipv4 $i" ;;
+		*:*:*) ipv6="$ipv6 $i" ;;
+		@*[a-zA-Z0-9]) ipset="$ipset $i" ;;
```

Comment:
I think you should include "-" and "_" as permitted set name characters.  They're nftables legal and I commonly use a construction "setname_4" and "setname_6" for ipv4/v6 specific sets.

Reply:
Good spot @ldir-EDB0, I've just committed some updates to the validations that should allow this now (as well as making them more robust overall).